src/community.rs

```diff
@@ -43,17 +43,17 @@ pub struct CommunityCard {
 
 impl Community {
     pub fn validate(&self) {
-        if self.name.len() > 30 {
-            panic!("Community name is limit to 30 characters");
+        if self.name.len() < 3 || self.name.len() > 30 {
```

Comment:
The proposed fix looks good to me, but here is a shorter way to check for range:

```suggestion
        if !(3..=30).contains(self.name.len()) {
```